src/Core/Web/XMLHttpRequestHelper.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generated On: 2015-10-5
+ * Class: XMLHttpRequestHelper. Simple promise-based wrapper around XMLHttpRequest
+ */
+
+
+define('Core/Web/XMLHttpRequestHelper', ['when'], function(when) {
+    return {
+        fetch: function(url, responseType) {
+            return when.promise(function(resolve, reject) {
+                var xhr = new XMLHttpRequest();
+                xhr.open('GET', url, true);
+                xhr.responseType = responseType;
+                xhr.crossOrigin = '';
+                xhr.onload = function(e) {
+                    resolve(this.response);
```

Comment:
Maybe returning a fetch-like response; or move to fetch (using [`whatwg-fetch`](https://github.com/github/fetch))